pcbnew2boardview.py

```diff
@@ -147,9 +158,19 @@ def main():
     parser.add_argument(
         "brd_file", metavar="BRD-FILE", type=argparse.FileType("wt"),
         help="output in .brd format")
-
+    
+    board = pcbnew.LoadBoard(args.kicad_pcb)
+    args = parser.parse_args()
+    convert(board, args.brd_file)
+    
+    parser = argparse.ArgumentParser(description="Generate an OBData file from a KiCad PCB")
```

Comment:
You should not be creating a second `ArgumentParser` since that's not how argparse works. Have you tested this functionality at all?

Reply:
Functionality is 100% but I'll get rid of this.

Reply:
No it's not 100%:

```
$ ./pcbnew2boardview.py --help
Traceback (most recent call last):
  File "/home/whitequark/Projects/kicad-boardview/./pcbnew2boardview.py", line 175, in <module>
    main()
  File "/home/whitequark/Projects/kicad-boardview/./pcbnew2boardview.py", line 160, in main
    board = pcbnew.LoadBoard(args.kicad_pcb)
                             ^^^^
UnboundLocalError: cannot access local variable 'args' where it is not associated with a value
```

Reply:
That's my bad, when I said functionality I meant the script's ability to output a `.brd` file and a `.obdata` file. I think I've fixed what you're talking about here though so please let me know if the most recent version is okay.

Reply:
I've done some more reading up on argparse, and I have a question about it.

Below is some code from `argparse.py` (link [here](https://github.com/python/cpython/blob/main/Lib/argparse.py#L1328-L1337))

```
    def __call__(self, string):
        # the special argument "-" means sys.std{in,out}
        if string == '-':
            if 'r' in self._mode:
                return _sys.stdin.buffer if 'b' in self._mode else _sys.stdin
            elif any(c in self._mode for c in 'wax'):
                return _sys.stdout.buffer if 'b' in self._mode else _sys.stdout
            else:
                msg = _('argument "-" with mode %r') % self._mode
                raise ValueError(msg)
```

This code makes me feel like an `r` character for `argparse.FileType` will be a read, but then any other character will make it a write. In the version of `pcbnew2boardview.py` included in this PR, I have

```
    parser.add_argument(
        "brd_file", metavar="BRD-FILE", type=argparse.FileType("wt"),
        help="output in .brd format")
    parser.add_argument(
        "obdata_file", metavar="OBDATA-FILE", type=argparse.FileType("w"),
        help="output in .obdata format")
```

The original one was `wt`, but I made mine `w` for no other reason than [this Stack Overflow answer](https://stackoverflow.com/a/16365493). My question is does it matter what this character is? Should they both be set to the same thing just for consistency?

This also affects `__init__.py` in the following lines.

```
    def Run(self):
        kicad_pcb = pcbnew.GetBoard()
        with open(kicad_pcb.GetFileName().replace('.kicad_pcb', '.brd'), 'wt') as brd_file:
            convert_brd(kicad_pcb, brd_file)
        with open(kicad_pcb.GetFileName().replace('.kicad_pcb', '.obdata'), 'w') as obdata_file:
            convert_obdata(kicad_pcb, obdata_file)
```

Reply:
`t` means text. This only changes things on Windows.

Reply:
I'm running it on Windows 11 23H2 without the t and it seems okay. Is it version specific? I can just leave it in if it's a compatibility thing.

Reply:
Ah, sorry, the part I neglected to mention is that `t` is the default, so `w` and `wt` are exactly equivalent. This is guaranteed by Python on all platforms. Feel free to use either `w` or `wt`, just be consistent.

Reply:
Okay! The script is all `w` in both files, is it okay to be merged?